app/routes/planet_routes.py

```diff
@@ -0,0 +1,73 @@
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request
+from .helpers import validate_planet
+
+planets_bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+# Get all planets
+@planets_bp.route("", methods=["GET"])
+def get_all_planets():
+    name_query = request.args.get("name")
+    id_query = request.args.get("id")
+    if name_query:
+        planets = Planet.query.filter_by(name = name_query)
+    elif id_query:
+        planets = Planet.query.filter_by(id = id_query)
+    else:
+        planets = Planet.query.all() 
+
+    planets_response = []
+    # planets = Planet.query.all()
+    for planet in planets:
+        planets_response.append(planet.to_json())
+    
+    return jsonify(planets_response),200
```

Comment:
👍  Great work filtering! We can also use list comprehension to build `planets_response`.

```python
planets_response = [planet.to_json() for planet in planets]
```